toml/src/main/kotlin/org/rust/toml/crates/local/CratesLocalIndexStartupActivity.kt

```diff
@@ -5,24 +5,50 @@
 
 package org.rust.toml.crates.local
 
+import com.intellij.openapi.application.ApplicationManager
 import com.intellij.openapi.project.Project
 import com.intellij.openapi.startup.StartupActivity
+import com.intellij.openapi.vfs.LocalFileSystem
+import com.intellij.openapi.vfs.VfsUtilCore
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.openapi.vfs.newvfs.BulkFileListener
+import com.intellij.openapi.vfs.newvfs.RefreshQueue
+import com.intellij.openapi.vfs.newvfs.events.VFileEvent
 import com.intellij.openapiext.isUnitTestMode
 import org.rust.cargo.project.model.cargoProjects
 import org.rust.ide.experiments.RsExperiments
 import org.rust.openapiext.isFeatureEnabled
+import java.io.File
+import java.nio.file.Paths
 
 /**
  * [CratesLocalIndexService] initializer.
- * Only tries to get service instance, without any updates to its crates index.
+ * Tries to get a service instance and binds crates local index update cargo registry index repository changes.
  */
 class CratesLocalIndexStartupActivity : StartupActivity.Background {
     override fun runActivity(project: Project) {
         if (isUnitTestMode) return
         if (!project.cargoProjects.hasAtLeastOneValidProject) return
 
         if (isFeatureEnabled(RsExperiments.CRATES_LOCAL_INDEX)) {
-            CratesLocalIndexService.getInstance()
+            val cratesService = CratesLocalIndexService.getInstance()
+
+            val userCargoRegistryIndexLocation = Paths.get(System.getProperty("user.home"), CratesLocalIndexServiceImpl.CARGO_REGISTRY_INDEX_LOCATION).toString()
+            LocalFileSystem.getInstance().addRootToWatch(userCargoRegistryIndexLocation, true)
+
+            val root = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(File(userCargoRegistryIndexLocation)) ?: return
```

Comment:
It is not obvious why `LocalFileSystem#addRootToWatch` does not work without these additional three lines of code. So I asked IntelliJ Platform team and here is what I learned.

VFS fills up lazily: any directory or file must be explicitly requested by the client to be included in VFS.

Here the `LocalFileSystem#refreshAndFindFileByIoFile` call requests the root directory. After that, the root directory is actually added to VFS (but without subdirectories). Another two lines request all the contents, and finally, we have everything included in VFS.

Another fact: to receive events about new subdirectories (created after our refresh), the client should explicitly request their contents. Fortunately, I suppose, we are not interested in new subdirectories here.

The reason why this routine is so complicated is that the main client of `LocalFileSystem#addRootToWatch` is the project model which maintains everything inside content roots by itself. So any other clients should do the same work as well.